src/peep/peep.c

```diff
@@ -3421,7 +3439,601 @@ static void peep_update_fixing(int steps, rct_peep* peep){
+		stationY += direction.y * 20;
+
+		peep->destination_x = stationX;
+		peep->destination_y = stationY;
+		peep->destination_tolerence = 2;
+	}
+
+	invalidate_sprite_2((rct_sprite *) peep);
+	if (peep_update_action(&x, &y, &tmp_xy_distance, peep) == 0) {
+		return true;
+	} else {
+		sprite_move(x, y, peep->z, (rct_sprite *) peep);
+		invalidate_sprite_2((rct_sprite *) peep);
+	}
+
+	return false;
+}
+
+/**
+ * rct2: 0x006C1504
+ */
+static bool peep_update_fixing_sub_state_13(bool firstRun, int steps, rct_peep *peep, rct_ride *ride) {
+	sint16 tmp_x, tmp_y, tmp_xy_distance;
+
+	if (!firstRun) {
+		if (peep->state == PEEP_STATE_INSPECTING) {
+			sub_6B7588(peep->current_ride);
+
+			peep->staff_rides_inspected++;
+			peep->window_invalidate_flags |= RIDE_INVALIDATE_RIDE_INCOME | RIDE_INVALIDATE_RIDE_LIST;
+
+			return true;
+		}
+
+		peep->staff_rides_fixed++;
+		peep->window_invalidate_flags |= RIDE_INVALIDATE_RIDE_INCOME | RIDE_INVALIDATE_RIDE_LIST;
+
+		peep->sprite_direction = peep->var_78 << 3;
+		peep->action = PEEP_ACTION_STAFF_ANSWER_CALL_2;
+		peep->action_frame = 0;
+		peep->action_sprite_image_offset = 0;
+
+		sub_693B58(peep);
+		invalidate_sprite_2((rct_sprite *) peep);
+	}
+
+	if (peep->action != 0xFF) {
+		peep_update_action(&tmp_x, &tmp_y, &tmp_xy_distance, peep);
+		return false;
+	}
+
+	ride_fix_breakdown(peep->current_ride, steps);
+
+	return true;
+}
+
+/**
+ * rct2: 0x006C157E
+ */
+static bool peep_update_fixing_sub_state_14(bool firstRun, rct_peep *peep, rct_ride *ride) {
+	sint16 x, y, xy_distance;
+
+	if (!firstRun) {
+		uint16 exitPosition = ride->exits[peep->current_ride_station];
+		if (exitPosition == 0xFFFF) {
+			exitPosition = ride->entrances[peep->current_ride_station];
+
+			if (exitPosition == 0xFFFF) {
+				peep_decrement_num_riders(peep);
+				peep->state = 0;
+				peep_window_state_update(peep);
+				return false;
+			}
+		}
+
+		uint16 exitX = (exitPosition & 0xFF) * 32;
+		uint16 exitY = (exitPosition >> 8) * 32;
+
+		exitX += 16;
+		exitY += 16;
+
+		rct_xy16 ebx_direction = word_981D6C[peep->var_78];
+		exitX -= ebx_direction.x * 19;
+		exitY -= ebx_direction.y * 19;
+
+		peep->destination_x = exitX;
+		peep->destination_y = exitY;
+		peep->destination_tolerence = 2;
+	}
+
+	invalidate_sprite_2((rct_sprite *) peep);
+	if (peep_update_action(&x, &y, &xy_distance, peep) == 0) {
+		peep_decrement_num_riders(peep);
+		peep->state = 0;
+		peep_window_state_update(peep);
+
+		return false;
+	}
+
+	uint16 z = ride->station_heights[peep->current_ride_station] * 8;
+
+	if (xy_distance >= 16) {
+		z += RCT2_ADDRESS(0x0097D21C, uint8)[ride->type * 8];
+	}
+
+	sprite_move(x, y, z, (rct_sprite *) peep);
+	invalidate_sprite_2((rct_sprite *) peep);
+
+	return false;
+}
+
+/**
+ * rct2: 0x6B7588
+ */
+static void sub_6B7588(int rideIndex) {
+	rct_ride *ride = get_ride(rideIndex);
+	ride->lifecycle_flags &= 0xFFFFFEFF;
+
+	ride->reliability += ((100 - ride->reliability) >> 2) * (scenario_rand() & 0xFF);
```

Comment:
not quite right ride->reliability needs to be `>>8` as you will notice the assembly instruction used is `bh`